core/src/main/java/org/apache/calcite/rel/core/Project.java

```diff
@@ -282,7 +282,12 @@ public RelWriter explainTerms(RelWriter pw) {
       if (!(exp.e instanceof RexInputRef)) {
         return null;
       }
-      mapping.set(((RexInputRef) exp.e).getIndex(), exp.i);
+
+      int source = ((RexInputRef) exp.e).getIndex();
+      if (mapping.getTargetOpt(source) != -1) {
+        return null;
```

Comment:
In a project, multiple source mapping to one target should never happen, i think this is an necessary check.

Reply:
> multiple source mapping to one target should never happen
I don't think so. plan like:
```
Project(MI=[$0], MA=[$0])
    TableScan(table=[[x, MYTABLE]], filters=[[=($0, 1)]])
```
is very common.

Reply:
What's the purpose of the change here? I see you already change to use getPartialMapping() in ProjectTableScanRule.

Reply:
Seems like there is no test case to cover this changes. The test case passed even without this change.

Reply:
It will throw exception wrongly without this change under some conditions . I will add a test case to cover it.